### ISSUE TITLE
Stop telling ArcMutex to be passed by reference

### ISSUE THREAD

**Issue by amab8901** [rust-lang/rust-clippy#15995]
### Summary

There's no advantage (AFAIK) with passing ArcMutex as reference rather than passing it as owned. If you need to use it in multiple places and don't want the borrow checker to complain, you just do `clone()` to create a new ArcMutex reference to the inner value inside ArcMutex.

### Lint Name

needless_pass_by_value

### Reproducer

I tried this code:

```rust
use std::sync::Arc;

use tokio::sync::Mutex;

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    let arg = Arc::new(Mutex::new(String::new()));

    lol(arg);
    Ok(())
}

pub(crate) fn lol(arg: Arc<Mutex<String>>) {
    println!("{arg:?}");
}

```

I saw this happen:

```
warning: this argument is passed by value, but not consumed in the function body
  --> src/main.rs:13:24
   |
13 | pub(crate) fn lol(arg: Arc<Mutex<String>>) {
   |                        ^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.91.0/index.html#needless_pass_by_value
   = note: `-W clippy::needless-pass-by-value` implied by `-W clippy::pedantic`
   = help: to override `-W clippy::pedantic` add `#[allow(clippy::needless_pass_by_value)]`
help: consider taking a reference instead
   |
13 | pub(crate) fn lol(arg: &Arc<Mutex<String>>) {
   |                        +

```

I expected to see this happen:
silent clippy

### Version

```text
rustc 1.91.0 (f8297e351 2025-10-28)
binary: rustc
commit-hash: f8297e351a40c1439a467bbbb6879088047f50b3
commit-date: 2025-10-28
host: x86_64-unknown-linux-gnu
release: 1.91.0
LLVM version: 21.1.2
```

**Comment by ada4a** [comment]
I think the replacement that the lint "has in mind" here is `Option<&Mutex<String>>`, but that does look a bit cursed.. Or is that what you meant with "passing ArcMutex as reference"?

**Comment by amab8901** [comment]
> I think the replacement that the lint "has in mind" here is `Option<&Mutex<String>>`, but that does look a bit cursed.. Or is that what you meant with "passing ArcMutex as reference"?

I realized that the Option is redundant for the point I'm trying to make. I updated the post by removing the option. 

Clippy is telling me to turn `Arc<Mutex<String>>` into `&Arc<Mutex<String>>`. But this doesn't make a lot of sense because ArcMutex is already a pointer, so clippy is basically telling me to make a pointer of a pointer

**Comment by Jarcho** [comment]
`needless_pass_by_value` is triggering correctly here. In general the solution is to use `&T` instead of `T` in order to avoid cloning, but different solutions are possible (e.g. `Arc<T>` to `&T` or `String` to `&str`). For `Arc` specifically, the clone can may be cheap, but it is not free. It can even become quite expensive if multiple threads are cloning it around the same time.

**Comment by amab8901** [comment]
`Arc<Mutex<T>>` has size 8. `&Arc<Mutex<T>>` has also size 8. Cloning and & both have size 8. There's no advantage with taking a reference over cloning. Why tell the user to do something that has no benefit?

**Comment by samueltardieu** [comment]
Cloning a `Arc` requires an exclusive access to the reference count to increment it which might introduce additional latency or bounce cache lines around in case of contention as written by Jarcho. Also, it includes an additional test on the count.

Taking a reference doesn't do anything and is much cheaper.
